[New Product] Amazon-Neptune

nadavgvili opened the pull request:
There is no AWS API to get End of life dates for RDS databases.

Better have this on an open-source platform than an individual web-scraper.

nadavgvili committed:
Create amazon-neptune.md

nadavgvili committed:
Update amazon-neptune.md

captn3m0 committed:
[aws-neptune] Minor rewording

captn3m0 commented on products/amazon-neptune.md:
```diff
@@ -0,0 +1,51 @@
+tags: amazon
+permalink: https://docs.aws.amazon.com/neptune/latest/userguide/engine-releases.html
+auto:
+-   custom: true
```
Since we currently don't have custom automation for AWS Neptune, are you planning to work on this? (See [eks.py](https://github.com/endoflife-date/release-data/blob/main/src/eks.py) for eg).

We not, we can drop the `auto` key for now, till we add this.
```suggestion
```

captn3m0 commented on products/amazon-neptune.md:
```diff
@@ -0,0 +1,65 @@
+---
+title: Amazon-Neptune
+category: db
```
We prefer using `service` for hosted services, and `db` for actual databases you can install and run on your own.

```suggestion
category: service
```

captn3m0 commented on products/amazon-neptune.md:
```diff
@@ -0,0 +1,65 @@
+>  Neptune powers graph use cases such as recommendation engines,
+>  fraud detection, knowledge graphs, drug discovery, and network security.
+
+Neptune engine versions almost always reach their end-of-life at the end of a calendar quarter. 
```
I've edited things slightly, but a short note on style: We prefer using [neutral third-person voice](https://github.com/endoflife-date/endoflife.date/wiki/Style-Guide#tone-and-text), which means avoiding words like "We/Our" (we aren't speaking on behalf of AWS), or "you" (instead preferring neutral terms like "users/customers", or avoiding them altogether).

Could you take a shot at fixing this? Happy to help if you have any questions.

captn3m0 replied:
Made some changes, closing as resolved

captn3m0 reviewed (changes requested):
Minor changes and suggestions.

captn3m0 committed:
[aws-neptune] front-matter improvements

captn3m0 committed:
[aws-neptune] table changes

Use a single row per minor version, since there are multiple
supported minor versions at the same time. This does create
some confusion regarding the releaseCycle being the same as the latest version, but I think that's okay and aligns with what the upstream now does.

Also add a new table for the suggested upgrade path for each minor version.

captn3m0 committed:
[neptune] Add versionCommand

Thought of using the curl command, but sticking to the AWS CLI
seems reasonable here.

captn3m0 committed:
[neptune] Tracking new versions

captn3m0 committed:
Update amazon-neptune.md

marcwrobel commented on products/amazon-neptune.md:
```diff
@@ -0,0 +1,97 @@
+    eol: 2023-01-30
+    latest: "1.0.5.1.R4"
+    latestReleaseDate: 2022-05-16
+    upgradeVersion: "1.1.0.0"
```
```suggestion
    upgradeVersion: "1.1.0.0"

```

marcwrobel commented on products/amazon-neptune.md:
```diff
@@ -0,0 +1,97 @@
+    aws neptune describe-db-clusters
+    --db-cluster-identifier your-neptune-db-identifier
+    --filters Name=engine,Values=neptune
+releases:
```
```suggestion

releases:
```

marcwrobel commented on products/amazon-neptune.md:
```diff
@@ -0,0 +1,97 @@
+>  that work with highly connected datasets. It supports multiple property-graph
+>  query languages - Apache TinkerPop Gremlin, openCypher, and SPARQL. 
+>  Neptune powers graph use cases such as recommendation engines,
+>  fraud detection, knowledge graphs, drug discovery, and network security.
```
```suggestion
> fully managed graph database service that makes it easy to build and run applications
> that work with highly connected datasets. It supports multiple property-graph
> query languages - Apache TinkerPop Gremlin, openCypher, and SPARQL. 
> Neptune powers graph use cases such as recommendation engines,
> fraud detection, knowledge graphs, drug discovery, and network security.
```

marcwrobel commented on products/amazon-neptune.md:
```diff
@@ -0,0 +1,97 @@
+> [Amazon Neptune](https://docs.aws.amazon.com/neptune/index.html) is a fast, reliable,
+>  fully managed graph database service that makes it easy to build and run applications
+>  that work with highly connected datasets. It supports multiple property-graph
+>  query languages - Apache TinkerPop Gremlin, openCypher, and SPARQL. 
```
```suggestion
>  query languages: Apache TinkerPop Gremlin, openCypher, and SPARQL. 
```

marcwrobel commented on products/amazon-neptune.md:
```diff
@@ -0,0 +1,97 @@
+---
+title: Amazon-Neptune
```
```suggestion
title: Amazon Neptune
```

marcwrobel commented on products/amazon-neptune.md:
```diff
@@ -0,0 +1,97 @@
+title: Amazon-Neptune
+category: service
+tags: amazon
+permalink: https://docs.aws.amazon.com/neptune/latest/userguide/engine-releases.html
```
This looks strange, think permalink and releasePolicyLink got mixed.

marcwrobel committed:
Reformat

marcwrobel committed:
fix metadata

marcwrobel reviewed (approved):
Made a few fix and verified the EOL dates (but not the latest versions - will automate it). LGTM now, will merge.

Thanks @nadavgvili for your contribution !
